### DISCUSSION TITLE
1D wave propagation

### DISCUSSION THREAD

**Discussion by tctheodosiou** [idaholab/moose#26740] [Q&A Modules: Solid mechanics] [answered]
Hello,
I ran a case, and didnt get the expected results. So, I tried to run a very simple case: 1D wave propagation; transient.
The case is the following: 1D bar, Length 1m, E=200GPA, fixed on the left, loaded on the right.
#=============-->
I applied a sinosoidal load F(t) = 0.01*sin(2*pi*t), for 0<t<0.5, and ran the simulation for 1s in 100 steps.

My problem is the following:
I open results in paraview, select the end point (the loaded one) and a point nead the middle of the bar. I expected a wave traveling from the left to the right, i.e. the midpoint should exhibit the same behavior as the loaded point, but with time-delay. Instead, the two points appear to be synced. 

Below is the code of my case. Initially, I used TensorMechanics/Master, and then DynamicsMaster, but no change:

```
[GlobalParams]
    displacements = disp_x
[]

[Mesh]
    [bar]
        type = GeneratedMeshGenerator
        dim = 1
        xmax = 1
        nx = 100
    []
[]

[Materials]
    [elasticity]
        type = ADComputeIsotropicElasticityTensor
        youngs_modulus = 200e9
        poissons_ratio = 0.3
    []

    [stress]
        type = ADComputeLinearElasticStress
    []

    [dens]
        type = ADDensity
        density = 6800
    []
[]

[BCs]
    [fix_left]
        type = ADDirichletBC
        variable = disp_x
        value = 0
        boundary = left
    []
[]

[DiracKernels]
    [load]
        type = FunctionDiracSource
        function = load
        variable = disp_x
        point = '1 0 0'
    []
[]

[Functions]
    [load]
      type = ParsedFunction
      expression = 'if(t<0.5,0.01*sin(2*pi*t), 0)'
    []
[]

[Modules/TensorMechanics/DynamicMaster]
    [all]
        add_variables = true
        strain = SMALL
        use_automatic_differentiation = true
    []
[]

[Executioner]
    type = Transient
    num_steps = 100
    end_time = 1
[]

[Outputs]
    exodus = true
[]
```

The displacement  I get (for the midpoint) is shown in the figure:
![2024-02-05_181229](https://github.com/idaholab/moose/assets/51261778/d7b10494-5958-4a56-8a61-2a8dfa1eb74a)
It shouldnt start moving at t=0.

What am I doing wrong? Must something very stuping that I am missing here! :(

**Comment by GiudGiud** [comment, marked as answer]
Hello

I think when you are NOT using dynamics, this is very normal. The regular Master tensor mechanics action is quasi-static, so the system always reaches equilibrium and that kind of perturbation would be instantaneous

For dynamics, I think:
- either your system is really rigid or your time steps are too big to see the delay
Can you try refining the time steps?

Guillaume

**Comment by tctheodosiou** [comment]
Changing to DynamicMaster and reducing the stiffness did the job. Thanks!
